### DISCUSSION TITLE
Where is the issue tracker for Copilot?

### DISCUSSION THREAD

**Discussion by johann-petrak** [community/community#47872] [Copilot Conversations] [answered]
### Select Topic Area

Product Feedback

### Body

After 2 minutes of using Copilot I hit the first really embarrassing fault: Python code that got inserted into VSCode used ¨ instead of " to quote strings, rendering all code having syntax errors. Really? 

Wanted to report that bug and could not find an issue tracker. Even discovering this discussions forum was not straightforward. Really?

OK, I have to admit I got angry spending 10 bucks for this already after 3 minutes: why would you not make it easy to give feedback and report bugs and problems with that product?

**Comment by msuliq** [comment, marked as answer]
Hi @johann-petrak, 
I don't think that GitHub Copilot has an issues tracker as in case with the usual GitHub public repositories, since there is no publicly available repository for it. I assume that the most effective way would be to search/create discussions in the GitHub community or in the broader web to see if someone has encountered such bug before or you would be a first one to mention a bug or an issue related to the Copilot.

Here is a link to a Copilot category in the Community Discussions: https://github.com/community/community/discussions/categories/copilot

Also, if you are unsatisfied with the quality of the service provided by the Copilot extension, you can always cancel your subscription and receive your money back minus the pro rata for the days passed after start of paid subscription (you won't be getting all money back, but only for the unused days left).

Please note that I am not affiliated with GitHub in any way and am responding to the best of my knowledge and personal experience.

Please mark this comment as accepted if it helped to answer your question. Thank you

**Reply by johann-petrak** [reply]
Pretty disappointing how GitHub  does not give proper feedback options for bug reports or feature requests.

**Reply by tcables** [reply]
this, exactly. we pay each month, and expect decent, and usual service which includes support

**Comment by iamcjwilliams** [comment]
@johann-petrak Our copilot team has been notified of the issue, Thank you for your feedback

**Reply by johann-petrak** [reply]
The odd thing is that copilot adapted after some time to suggesting the proper quotes and now it works fine. Still odd that it came up with the wrong quote character initially (I am not aware of any programming language which uses that quite character).

Apart from this, I do find it disappointing how hard it is to give direct feedback to GitHub, not just on this but on many other issues where the community has been asking for changes, features or similar. It really reminds one of the attitude of Microsoft, which is the opposite of most open source projects GitHub hosts where issue trackers and a feedback culture are part of the philosophy.

**Reply by JeckWild-JWS** [reply]
In my case it's similar, it's opening a new bracket for a method that already has a closing bracket, and a "return" keyword. Feels like Copilot is not scanning ahead very much. 
<img width="525" alt="Untitled" src="https://user-images.githubusercontent.com/111518424/226713879-fcaec37d-03c1-4a64-af7c-93eb19025a77.png">

**Reply by johann-petrak** [reply]
Yes found all kinds of other weird issues in the meantime but I am not bothering any more trying to give feedback, because quite frankly, I find the lack of a proper issue tracker or issue feedback channel and the way how GitHub does not provide any of the options for its own software (Copilot or even GitHub itself) which are just standard in all the projects GitHub makes their money on just frustrating and stupid. 
Kind of microsoftesque.

**Reply by tcables** [reply]
I second this.

**Reply by ayim** [reply]
@JeckWild-ePS we've put work into the bracket completion logic for GitHub Copilot in Visual Studio.
All issues related to the Copilot extensions in VS are tracked publicly through the VS Feedback tool:
https://developercommunity.visualstudio.com/VisualStudio?q=Copilot

**Comment by sovetski** [comment]
I agree with this topic, if we can't suggest improvements, how is this product supposed to improve itself? For a paid system, we should have at least the right to that...

**Reply by kenny1983** [reply]
While I am also disappointed by the lack of an official issue tracker, there is [this discussion](https://github.com/orgs/community/discussions/6612) that outlines what seems to be the closest thing to a proper issue reporting procedure that currently exists.

Basically just create a discussion here in these forums and report your issue there. But also, being an AI-driven piece of software, I would imagine that correcting mistakes in any code that Copilot gives you would automatically teach it (or at least help to teach it) the proper way to do things in that particular scenario.

**Comment by GurkeSchaleSammler** [comment]
Hello man, The Pilot has many "Rassist and Right-orientated", for example : "The Jew need " (if i want to spell need to eat less on their holidays) and he spells " s to be killed/ gassed" or same shit for rassist need, then he says need to get more supporters etc. its not so good.....

**Comment by ayim** [comment]
For general GitHub Copilot issues, this is the correct forum.
For Visual Studio, I recommend using the "Report a Problem" feature within the IDE.
![image](https://github.com/community/community/assets/10951869/887ea53d-cafc-41de-a15b-2943783eb6b3)
